app/lib/domain/controllers/tts_controller.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_tts/flutter_tts.dart';
+import 'package:logger/logger.dart';
+
+part 'package:word_link/infrastructure/tts_repository.dart';
+
+abstract class TtsController {
+  static TtsController? _instance;
+
+  static TtsController get instance => _instance ??= _TtsRepository();
+
+  late FlutterTts? _flutterTts;
+  final _logger = Logger();
```

Comment:
Don't create a new instance of logger, we have one for all the app.
Simply use `logger.i` or `logger.e` when needed